javalib/src/main/scala/java/nio/Buffer.scala

```diff
@@ -1,6 +1,14 @@
 package java.nio
 
 abstract class Buffer private[nio] (val _capacity: Int) {
+  private[nio] type ElementType
+
+  private[nio] type BufferType >: this.type <: Buffer {
+    type ElementType = Buffer.this.ElementType
+  }
```

Comment:
We need this because some methods, such as one of the overloads of `put`, take a *parameter* of the same kind of buffer.